cargo-dist/src/backend/installer/docker.rs

```diff
@@ -0,0 +1,203 @@
+//! CI script generation
+//!
+//! In the future this may get split up into submodules.
+
+use std::process::Command;
+
+use axoasset::LocalAsset;
+use camino::Utf8PathBuf;
+use cargo_dist_schema::DistManifest;
+use serde::Serialize;
+
+use crate::{
+    backend::templates::{Templates, TEMPLATE_INSTALLER_DOCKER},
+    config::DependencyKind,
+    errors::{DistError, DistResult},
+    DistGraph, ReleaseIdx,
+};
+
+/// Info needed to build an msi
+#[derive(Debug, Clone)]
+pub struct DockerInstallerInfo {
+    /// Target triple this image is for
+    pub target: String,
+    /// Release this is for
+    pub release_idx: ReleaseIdx,
+    /// What to tag the image as
+    pub tag: String,
+    /// Binaries we'll be baking into the docker image
+    pub bins: Vec<String>,
+    /// Final file path of the docker image
+    pub file_path: Utf8PathBuf,
+    /// Dir stuff goes to
+    pub package_dir: Utf8PathBuf,
+    /// Additional dirs/files to include
+    pub includes: Vec<Utf8PathBuf>,
+}
+
+/// Info about running cargo-dist in Github CI
+#[derive(Debug, Serialize)]
+pub struct DockerfileInfo {
+    /// List of binaries this docker image should include
+    bins: Vec<Bin>,
+    /// Info about the runner image
+    runner: RunnerImage,
+}
+
+/// A binary
+#[derive(Debug, Serialize)]
+struct Bin {
+    /// Path to where the binary should be copied from, on the host system
+    source: String,
+    /// Name of the binary, as exposed to users of the image
+    name: String,
+}
+
+/// Info about the runner image
+#[derive(Debug, Serialize)]
+struct RunnerImage {
+    /// Base image (e.g. some debian distro)
+    image: String,
+    /// Apt deps to install
+    apt_deps: Vec<String>,
+    /// Additional files/dirs to copy into the image (e.g. examples)
+    includes: Vec<Utf8PathBuf>,
+}
+
+/// As of this rewriting this is the latest debian release
+const RUNNER_IMAGE: &str = "debian:bookworm-slim";
```

Comment:
Might be worth trying to sync this with the builder at this point, instead of using a different Debian flavour. In particular, since you're making use of apt dependency versions, there's a very real chance the specific versions won't line up between the Ubuntu builder and this version of Debian.